contracts/lockdrop/src/contract.rs

```diff
@@ -2024,6 +2035,57 @@ pub fn query_lockup_info(
     })
 }
 
+/// @dev Returns pending asset rewards regarding the user with
+pub fn query_pending_asset_reward(
+    deps: Deps,
+    env: Env,
+    recipient: String,
+    terraswap_lp_token: String,
+    duration: u64,
+) -> StdResult<PendingAssetRewardResponse> {
+    let recipient = addr_validate_to_lower(deps.api, &recipient)?;
+    let terraswap_lp_token = addr_validate_to_lower(deps.api, &terraswap_lp_token)?;
+
+    let lockup_key = (&terraswap_lp_token, &recipient, U64Key::new(duration));
+
+    let lockup_info_opt = LOCKUP_INFO
+        .may_load(deps.storage, lockup_key.clone())?
+        .filter(|lock_info| lock_info.astroport_lp_transferred.is_none());
+
+    let mut user_reward = Uint128::zero();
+    if let Some(lockup_info) = lockup_info_opt {
+        let pool_info = ASSET_POOLS.load(deps.storage, &terraswap_lp_token)?;
+        if !pool_info.has_asset_rewards {
+            return Err(StdError::generic_err("This pool does not have rewards"));
+        }
+
+        let MigrationInfo {
+            astroport_lp_token, ..
+        } = pool_info
+            .migration_info
+            .as_ref()
+            .expect("Terraswap liquidity hasn't migrated yet!");
+
+        let pending_rewards: Asset = deps.querier.query_wasm_smart(
+            astroport_lp_token,
+            &astroport::pair_stable_bluna::QueryMsg::PendingReward {
+                user: env.contract.address.to_string(),
+            },
+        )?;
+
+        let reward_index = Decimal256::from_ratio(
+            Uint256::from(pending_rewards.amount),
+            pool_info.terraswap_amount_in_lockups,
+        );
+
+        user_reward = (reward_index * Uint256::from(lockup_info.lp_units_locked)).try_into()?;
```

Comment:
Here we also need to add the difference between indexes https://github.com/astroport-fi/astroport-bootstrapping/blob/74032d5e11dfa823db291fbafc57ff82273aba5b/contracts/lockdrop/src/state.rs#L20

Reply:
Done, thaks

Reply:
TOTAL_ASSET_REWARD_INDEX - USERS_ASSET_REWARD_INDEX

Reply:
Changed

Reply:
Let's initialize TOTAL_ASSET_REWARD_INDEX as [here](https://github.com/astroport-fi/astroport-periphery/blob/b341863bb3e501539f8990656d1fc0364e93f411/contracts/lockdrop/src/contract.rs#L1723) and use [this function](https://github.com/astroport-fi/astroport-periphery/blob/b341863bb3e501539f8990656d1fc0364e93f411/contracts/lockdrop/src/contract.rs#L1756) to calculate pending rewards

Reply:
Changed